library/xml-model/src/main/java/org/cru/godtools/xml/model/Fallback.kt

```diff
@@ -0,0 +1,33 @@
+package org.cru.godtools.xml.model
+
+import androidx.annotation.RestrictTo
+import org.cru.godtools.xml.XMLNS_CONTENT
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+
+class Fallback : Content, Parent {
+    companion object {
+        internal const val XML_FALLBACK = "fallback"
+    }
+
+    override val content: List<Content>
+
+    internal constructor(parent: Base, parser: XmlPullParser) : super(parent, parser) {
+        when (parser.name) {
+            Paragraph.XML_PARAGRAPH -> {
+                parser.require(XmlPullParser.START_TAG, XMLNS_CONTENT, Paragraph.XML_PARAGRAPH)
+                if (parser.getAttributeValue(null, Paragraph.XML_FALLBACK)?.toBoolean() != true) {
+                    throw XmlPullParserException("expected fallback=\"true\" at ${parser.positionDescription}")
+                }
+            }
+            else -> parser.require(XmlPullParser.START_TAG, XMLNS_CONTENT, XML_FALLBACK)
+        }
+
+        content = parseContent(parser)
+    }
+
+    @RestrictTo(RestrictTo.Scope.TESTS)
+    constructor(parent: Base, content: ((Fallback) -> List<Content>?)? = null) : super(parent) {
+        this.content = content?.invoke(this).orEmpty()
+    }
```

Comment:
Is this deciding which parser to use based on the name?

Reply:
we are using an `XmlPullParser`, so there is a `parser` object that processes an XML stream and exposes encountered nodes via methods on the `parser` object.

what the first constructor is doing is taking in the parser object, requiring that it is currently processing either `<content:paragraph fallback="true">` or `<content:fallback>`, then calls `parseContents()` to actually parse any child content.

The second constructor is meant for tests and allows us to completely bypass the xml parsing logic and build a `Fallback` node with the related objects we want.